mobile/android/android-emulator-services.ts

```diff
@@ -320,22 +334,18 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 		}).future<boolean>()();
 	}
 
+	public getAllRunningEmulators(): IFuture<string[]> {
+		return ((): string[] => {
+			let outputRaw: string[] = this.$childProcess.execFile(this.adbFilePath, ['devices']).wait().split(EOL);
+			let emulators = this.getRunningAvdEmulators(outputRaw).wait().concat(this.getRunningGenymotionEmulators(outputRaw).wait());
+			return emulators;
+		}).future<string[]>()();
+	}
+
 	private getRunningEmulators(): IFuture<string[]> {
 		return (() => {
-
-			let emulatorDevices: string[] = [];
 			let outputRaw: string[] = this.$childProcess.execFile(this.adbFilePath, ['devices']).wait().split(EOL);
-			if(this.$options.geny) {
-				emulatorDevices = this.getRunningGenymotionEmulators(outputRaw).wait();
-			} else {
-				_.each(outputRaw, (device: string) => {
-					let rx = device.match(AndroidEmulatorServices.RUNNING_ANDROID_EMULATOR_REGEX);
-
-					if(rx && rx[1]) {
-						emulatorDevices.push(rx[1]);
-					}
-				});
-			}
+			let emulatorDevices = this.$options.geny ? this.getRunningGenymotionEmulators(outputRaw).wait() : this.getRunningAvdEmulators(outputRaw).wait();
```

Comment:
Not a merge stopper, but I'd suggest to rewrite this as an if() else .. else { throw /assert } to help us when we add Visual Studio emulator.